OpenRA.Game/Traits/SelectionDecorations.cs

```diff
@@ -78,6 +78,13 @@ IEnumerable<IRenderable> DrawPips(WorldRenderer wr, Actor self, int2 basePositio
 			if (!pipSources.Any())
 				yield break;
 
+			var isAlly = self.Owner.IsAlliedWith(self.World.LocalPlayer)
```

Comment:
Sorry, I wasn't really clear with my previous comment: These checks need to be against self.World.RenderPlayer in order for it to behave correctly with replays and the disable shroud cheat.